File: js/helloworld/lib/HelloWorld.js
'use strict';

var AbstractComponent = require('kevoree-entities/lib/AbstractComponent');

/**
 * Prints &#34;Hello World!&#34; and &#34;Bye World!&#34;
 *
 * @type {HelloWorld}
 */
module.exports = AbstractComponent.extend({
  toString: 'HelloWorld',
  tdef_version: 1,

  /**
   * Called when your component starts
   *
   * @param {Function}     done call this when you are done starting
   *                            done(): all good
   *                            done(error): something went wrong
   */
  start: function (done) {
    this.log.info(this.toString(), 'Hello World!');
    done();
  },

  /**
   * Called when your component stops
   *
   * @param {Function}     done call this when you are done stopping
   *                            done(): all good
   *                            done(error): something went wrong
   */
  stop: function (done) {
    this.log.info(this.toString(), 'Bye World!');
    done();
  }
});

File: java/helloworld/src/main/java/my/company/kevoree/HelloWorld.java
package my.company.kevoree;

import org.kevoree.annotation.ComponentType;
import org.kevoree.annotation.Start;
import org.kevoree.annotation.Stop;

/**
 *
 * Created by leiko on 12/2/16.
 */
@ComponentType(version = 1)
public class HelloWorld {

    @Start
    public void start() {
        System.out.println("Hello World!");
    }

    @Stop
    public void stop() {
        System.out.println("Bye World!");
    }
}
